.github/workflows/enos-run.yml

```diff
@@ -273,9 +276,18 @@ jobs:
           ENOS_VAR_test_email: ${{ secrets.SERVICE_USER_EMAIL }}
           ENOS_VAR_boundary_docker_image_name: ${{ inputs.docker-image-name }}
           ENOS_VAR_boundary_docker_image_file: ./support/boundary_docker_image.tar
+          ENOS_DEBUG_DATA_ROOT_DIR: ./support/debug-data
         run: |
           export ENOS_VAR_enos_user=$GITHUB_ACTOR && \
           enos scenario run --timeout 60m0s --chdir ./enos ${{ matrix.filter }}
+      - name: Upload Debug Data
+        if: failure()
```

Comment:
Sometimes, other parts of the workflow fail, which would unnecessarily trigger this. In other similar cases, I had to use something like
```
if: ${{ always() && steps.run_retry.outcome == 'failure' }}
```

Reply:
good suggestion, I like this.